code/game/objects/machinery/telecomms/broadcasting.dm

```diff
@@ -199,6 +199,25 @@
 			continue
 		hearer.Hear(rendered, virt, language, message, frequency, spans)
 
+
+	//Now we can handle radio TTS for squads
+
+	var/mob/living/carbon/human/speaker = virt.source
+	if(speaker.voice && !is_banned_from(speaker.ckey, "TTS"))
+		var/headset_target = HEADSET_TTS_ALL
+		if((FREQ_ALPHA < frequency && frequency < FREQ_CUSTOM_SQUAD_MAX) || (FREQ_ZULU < frequency && frequency < MAX_ERT_FREQ))
+			headset_target = HEADSET_TTS_SQUAD
+			if(speaker?.assigned_squad.squad_leader == speaker)
+				headset_target = HEADSET_TTS_SL_ONLY //Lower the bar if the speaker is the active aSL
+
+		if(speaker in receive)
+			receive -= speaker //This list isn't used again, so we can just cut out the original speaker from it so TTS doesn't play twice
+
+		for(var/mob/living/carbon/human/potential_hearer in receive)
+			var/obj/item/radio/headset/radio = potential_hearer.wear_ear
+			if(radio.squad_tts_mode >= headset_target && potential_hearer.stat < UNCONSCIOUS && potential_hearer.client?.prefs.muted & MUTE_TTS)
```

Comment:
this creations a zillion runtimes when TTS is not on. Probably on other places as well I assume.